tests/functional/aws-node-sdk/test/bucket/getCorsResponse.js

```diff
@@ -0,0 +1,1083 @@
+                        `Error: ${key} should not have value`);
+                });
+            }
+            return callback();
+        });
+    });
+
+    req.on('error', err => {
+        process.stdout.write('err sending request');
+        return callback(err);
+    });
+    req.end();
+}
+
+describe('OPTIONS request without bucket', () => {
+    it('should respond no such bucket if bucket does not exist', done => {
+        const headers = {
+            Origin: 'http://www.allowedwebsite.com',
+        };
+        methodRequest({ method: 'GET', headers, code: 'NoSuchBucket',
+        headersResponse: null }, done);
+    });
+    it('should return BadRequest for OPTIONS request without origin', done => {
+        const headers = {};
+        methodRequest({ method: 'OPTIONS', headers, code: 'BadRequest',
+        headersResponse: null }, done);
+    });
+    it('should return BadRequest for OPTIONS request without ' +
+    'Access-Control-Request-Method', done => {
+        const headers = {
+            Origin: 'http://www.allowedwebsite.com',
+        };
+        methodRequest({ method: 'OPTIONS', headers, code: 'BadRequest',
+        headersResponse: null }, done);
+    });
+});
+
+describe('OPTIONS requests with existing bucket', () => {
+    beforeEach(done => {
+        s3.createBucket({ Bucket: bucket, ACL: 'public-read' }, err => {
+            _waitForAWS(done, err);
+        });
+    });
+    afterEach(done => {
+        s3.deleteBucket({ Bucket: bucket }, err => {
+            _waitForAWS(done, err);
+        });
+    });
+
+    it('should allow GET without CORS', done => {
+        const headers = {
+            Origin: 'http://www.allowedwebsite.com',
+        };
+        methodRequest({ method: 'GET', headers, code: 200,
+        headersResponse: null }, done);
+    });
+    it('should allow HEAD without CORS', done => {
+        const headers = {
+            Origin: 'http://www.allowedwebsite.com',
+        };
+        methodRequest({ method: 'HEAD', headers, code: 200,
+        headersResponse: null }, done);
+    });
+    it('should return AccessForbidden for OPTIONS request on bucket without ' +
+    'CORSConfiguration', done => {
+        const headers = {
+            'Origin': 'http://www.allowedwebsite.com',
+            'Access-Control-Request-Method': 'GET',
+        };
+        methodRequest({ method: 'OPTIONS', headers, code: 'AccessForbidden',
+        headersResponse: null }, done);
+    });
+
+    describe('allow PUT, POST, DELETE, GET methods and allow only ' +
+    'one origin', () => {
+        const corsParams = {
+            Bucket: bucket,
+            CORSConfiguration: {
+                CORSRules: [
+                    {
+                        AllowedMethods: [
+                            'PUT', 'POST', 'DELETE', 'GET',
+                        ],
+                        AllowedOrigins: [
+                            'http://www.allowedwebsite.com',
```

Comment:
might as well use a variable for http://www.allowedwebsite.com since it is used so much